packages/process/src/common/shell-quoting.spec.ts

```diff
@@ -0,0 +1,176 @@
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { expect } from 'chai';
+
+import { escapeForShell, BashQuotingFunctions, ShellQuoting, CmdQuotingFunctions, PowershellQuotingFunctions } from './shell-quoting';
+
+describe('Shell arguments escaping:', () => {
+
+    // Procedurally execute tests from this list of data.
+    const testData = {
+        // https://www.gnu.org/software/bash/manual/html_node/Quoting.html
+        bash: {
+            quotingFunctions: BashQuotingFunctions,
+            data: {
+                [ShellQuoting.Escape]: [
+                    { input: 'abc', expected: 'abc' },
+                    { input: 'ab c', expected: 'ab\\ c' },
+                    { input: 'ab"c', expected: 'ab\\"c' },
+                    { input: 'ab\'c', expected: 'ab\\\'c' },
+                    { input: 'ab\\ c\\', expected: 'ab\\\\\\ c\\\\' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: 'setTimeout\\(\\(\\)\\ =\\>\\ \\{\\ console.log\\(1,\\ \\"2\\\'3\\"\\)\\;\\ \\},\\ 100\\)',
+                    },
+                ],
+                [ShellQuoting.Strong]: [
+                    { input: 'abc', expected: '\'abc\'' },
+                    { input: 'ab c', expected: '\'ab c\'' },
+                    { input: 'ab"c', expected: '\'ab"c\'' },
+                    { input: 'ab\'c', expected: '\'ab\'"\'"\'c\'' },
+                    { input: 'ab\\ c\\', expected: '\'ab\\ c\\\'' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: '\'setTimeout(() => { console.log(1, "2\'"\'"\'3"); }, 100)\'',
+                    },
+                ],
+                [ShellQuoting.Weak]: [
+                    { input: 'abc', expected: '"abc"' },
+                    { input: 'ab c', expected: '"ab c"' },
+                    { input: 'ab"c', expected: '"ab\\"c"' },
+                    { input: 'ab\'c', expected: '"ab\'c"' },
+                    { input: 'ab\\ c\\', expected: '"ab\\ c\\\\"' },
+                    {
+                        input: 'setTimeout(() => { console.log(1, "2\'3"); }, 100)',
+                        expected: '"setTimeout(() => { console.log(1, \\"2\'3\\"); }, 100)"',
+                    },
+                ]
+            },
+        },
+        // https://ss64.com/nt/syntax-esc.html ?
```

Comment:
I don't understand the following comment or the need for `?`.

Reply:
These are simply references, might be handy to understand the tests. Removed the "?".